tchannel/tornado/peer.py

```diff
@@ -608,13 +664,28 @@ def add(self, peer):
 
         if isinstance(peer, basestring):
             # Assume strings are host-ports
-            peer = self.peer_class(self.tchannel, peer)
+            peer = self.peer_class(
+                tchannel=self.tchannel,
+                hostport=peer,
+                on_conn_change=self.update_heap,
+            )
+            peer.rank = self.rank_calculator.get_rank(peer)
```

Comment:
You probably need to set the rank on peers that you're not constructing on demanding but adding manually as well, don't you?